fixtures/provider.py

```diff
@@ -58,6 +58,15 @@
 SETUP_FAIL_LIMIT = 3
 
 
+def pytest_addoption(parser):
+    # Create the cfme option group for use in other plugins
+    parser.getgroup('cfme')
+    parser.addoption("--provider-limit", action="store", default=0, type=int,
+        help=(
+            "Number of cloud/infra providers allowed to coexist on appliance. 0 means no limit. "
```

Comment:
Unless you turn the `list_providers(...)` below into `list_providers_by_class(CloudInfraProvider, use_global_filter=False)`. Then it would be just the two.